integration-tests/src/main/java/org/apache/cassandra/sidecar/testing/SharedClusterSidecarIntegrationTestBase.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sidecar.testing;
+
+import io.vertx.core.net.JksOptions;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.client.WebClientOptions;
+
+import static io.vertx.core.Vertx.vertx;
+import static org.apache.cassandra.sidecar.testing.MtlsTestHelper.EMPTY_PASSWORD_STRING;
+
+/**
+ * Builds on top of {@link SharedClusterIntegrationTestBase} and adds functionality to interact
+ * with the Sidecar process with a trusted client
+ */
+public abstract class SharedClusterSidecarIntegrationTestBase extends SharedClusterIntegrationTestBase
```

Comment:
Is it a mistake that this test class is placed under `main`? If so, please move it to test scope.